doc: add document for dashboards assistant (#8622)
opensearch-project/documentation-website: _dashboards/dashboards-assistant/data-summary.md

---
layout: default
title: Data summary
parent: OpenSearch Assistant for OpenSearch Dashboards
nav_order: 1
has_children: false
---

# Data summary

> This is an experimental feature and is not recommended for use in a production environment. For updates on the progress the feature or if you want to leave feedback, join the discussion in the [OpenSearch forum](https://forum.opensearch.org/).    

Reviewer: ```suggestion
This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, join the discussion on the [OpenSearch forum](https://forum.opensearch.org/).    
```

{: .warning}

The OpenSearch Dashboards Assistant data summary uses large language models (LLMs) to help you generate summaries for data stored in OpenSearch indexes. This tool provides an efficient way to gain insights from large datasets, making it easier to understand and act on the information contained in your OpenSearch indexes.

## Configuration

> To configure data summary, use the following steps.

Reviewer: ```suggestion
To configure the data summary feature, use the following steps.
```

> 

Reviewer: Global: Either "the data summary feature" or "data summarization"

### Prerequisite

> Before using data summary, enable query enhancements in OpenSearch Dashboards as follows:

Reviewer: ```suggestion
Before using the data summary feature, enable query enhancements in OpenSearch Dashboards as follows:
```

1. On the top menu bar, go to **Management > Dashboards Management**. 

> 1. In the left navigation, select **Advanced settings**.

Reviewer: Should "pane" follow "navigation"?

> 1. In the left navigation, select **Advanced settings**.

Reviewer 2: ```suggestion
1. In the left navigation pane, select **Advanced settings**.
```

1. On the settings page, toggle **Enable query enhancements** to **On**.

> ### Step 1: Enable data summary

Reviewer: ```suggestion
### Step 1: Enable the data summary feature
```

> To enable data summary, configure the following `opensearch_dashboards.yml` setting:

Reviewer: ```suggestion
To enable the data summary feature, configure the following `opensearch_dashboards.yml` setting:
```

```yaml
queryEnhancements.queryAssist.summary.enabled: true
```
{% include copy.html %}

### Step 2: Create a data summary agent

> To orchestrate data summary, create an `os_query_assist_ppl` agent. To create an agent, send a `POST /_plugins/_ml/agents/_register` request and provide the agent template as a payload:

Reviewer: ```suggestion
To orchestrate data summarization, create an `os_query_assist_ppl` agent. To create an agent, send a `POST /_plugins/_ml/agents/_register` request and provide the agent template as a payload:
```

    "sample_count":1,
    "total_count":383,
    "question":"Are there any errors in my logs?",
    "ppl":"source=opensearch_dashboards_sample_data_logs| where QUERY_STRING(['response'], '4* OR 5*')"
}
```
{% include copy-curl.html %}

The following table describes the Assistant Data Summary API parameters.

Parameter | Required/Optional | Description 
:--- | :--- | :---

> `sample_data` | Required | A sample of data returned by the specified query, used as input for summarization.

Reviewer: ```suggestion
`sample_data` | Required | A sample of data returned by the specified query and used as input for summarization.
```

`question` | Optional | The user's natural language question about the data, which guides the summary generation.

> `ppl` | Optional | The Piped Processing Language (PPL) query to retrieve data; in query assistance, this is generated by the LLM from the user's natural language question.

Reviewer: ```suggestion
`ppl` | Optional | The Piped Processing Language (PPL) query used to retrieve data; in query assistance, this is generated by the LLM using the user's natural language question.
```

> `sample_count` | Optional | The number of entries included in sample_data.

Reviewer: Above: The meaning of "in query assistance" is unclear.

`total_count` | Optional | The total number of entries in the full query result set.

## Viewing data summaries in OpenSearch Dashboards

To view alert insights in OpenSearch Dashboards, use the following steps:

1. On the top menu bar, go to **OpenSearch Dashboards > Discover**.

> 1. In the query language dropdown list, select **PPL**. You will see the generated data summary below the query text, as shown in the following image.

Reviewer: ```suggestion
1. From the query language dropdown list, select **PPL**. You will see the generated data summary after the query text, as shown in the following image.
```
